Add CEReactions tests for HTMLBaseElement

xiuqijix opened the pull request:
part of issue #5727

xiuqijix committed:
Add CEReactions tests for HTMLBaseElement

kaixinjxq commented:
@rniwa, would you like to help to review this PR?

rniwa commented on custom-elements/reactions/HTMLBaseElement.html:
```diff
@@ -0,0 +1,45 @@
+testReflectAttribute('href', 'href', '/', '/', 'http://example.com/', 'http://example.com/', 'href on HTMLBaseElement', 'base', HTMLBaseElement);
+testReflectAttribute('target', 'target', '_blank', '_blank', '_self', '_self', 'target on HTMLBaseElement', 'base', HTMLBaseElement);
+
+function testReflectAttribute(jsAtName, coAtName, jsAtValue1, coAtValue1, jsAtValue2, coAtValue2, name, elementName, interfaceName) {
```
This function should probably be added to custom-elements-helpers.js instead.

kaixinjxq replied:
@rniwa, many thanks for your review, I think [reactions.js](https://github.com/web-platform-tests/wpt/blob/master/custom-elements/reactions/resources/reactions.js) is more appropriate, and I've addressed it in latest commit and replaced the `testReflectAttrWithParentNode` function in [HTMLButtonElement.html](https://github.com/web-platform-tests/wpt/blob/master/custom-elements/reactions/HTMLButtonElement.html) as well, PTAL.

xiuqijix committed:
Create common function to reactions.js

- move testReflectAttribute function to reactions.js
- replace testReflectAttrWithParentNode in HTMLButtonElement.html

foolip commented:
This PR is blocked on the required "Travis CI - Pull Request" check after https://github.com/web-platform-tests/wpt/issues/14499. In order to trigger it, I will close and reopen this PR.

xiuqijix committed:
Replace pElementNameFunction with getParentElement

kaixinjxq commented:
@rniwa, I'm working on another `CEReactions` tests that may base on this new common function, could you please check if this PR is fine to you?

xiuqijix committed:
Modify the tests for boolean attributes

kaixinjxq commented:
@rniwa, friendly ping.

domenic commented:
LGTM but needs rebasing. It may also be nicer to do separate PRs to fix existing tests, instead of putting that work in this PR.

xiuqijix committed:
Revert "Modify the tests for boolean attributes"

This reverts commit 60d219be5d5bbd63575e989f27a24f3e246b53a0.

kaixinjxq commented:
@domenic, many thanks for your review! I just rebase this PR and revert the commit "Modify the tests for boolean attributes", I will address it in a separate PR.
